### PR TITLE
 As a user i want to have take profit order so i can take my profit

### PR DESCRIPTION
**Pull request by TymofiiSavinov**

### CODE REVIEW

**Comment by markblaize** [comment]
Task linked: [CU-37q2wxt As a User I want to have take-profit order so I can take my profit](https://app.clickup.com/t/37q2wxt)

File: contracts/leverage_trading/src/view.rs [hunk 1]
```diff
@@ -433,6 +434,20 @@ impl Contract {
...
         }
     }
+
+    pub fn take_profit_order_view(&self, order_id: U128) -> TakeProfitOrderView {
+        require!(
+            Some(signer_account_id()) == self.get_account_by(order_id.0),
+            "You have no access for this order."
+        );
+
+        let tpo = self.take_profit_orders.get(&(order_id.0 as u64)).unwrap();
```
Comment (arsenijkovalov) [inline contracts/leverage_trading/src/view.rs:444]: Replace `.unwrap()` with panic with appropriate message

File: contracts/leverage_trading/src/create_order.rs [hunk 2]
```diff
@@ -251,6 +250,103 @@ impl Contract {
         let order: Order = serde_json::from_str(order.as_str()).unwrap();
         self.add_or_update_order(&account_id, order, order_id);
     }
+
+    #[payable]
+    pub fn create_take_profit_order(
```
Comment (arsenijkovalov) [inline contracts/leverage_trading/src/create_order.rs:255]: Should verify:
1. if User owns the position and this position is opened (partially implemented)
2. if no take-profit orders exist

File: contracts/leverage_trading/src/execute_order.rs [hunk 3]
```diff
@@ -86,12 +111,19 @@ impl Contract {
...
         order_id: U128,
+        take_profit_order: bool,
     ) -> PromiseOrValue<U128> {
         if !is_promise_success() {
             panic!("Some problem with remove liquidity");
         } else {
             self.mark_order_as_executed(order, order_id);
 
+            if !take_profit_order {
+                if let Some(tpo) = self.take_profit_orders.get(&(order_id.0 as u64)) {
```
[the remark this answers is not in the document]
Comment (TymofiiSavinov) [reply contracts/leverage_trading/src/execute_order.rs:122]: done

**Review by arsenijkovalov — APPROVED** [review]

**Pull request merged**